src/main/kotlin/racingcar/domain/Race.kt

```diff
@@ -0,0 +1,29 @@
+package racingcar.domain
+
+class Race(private val raceSetting: RaceSetting) {
+
+    private val cars: List<Car> = List(raceSetting.numberOfCar) { index ->
+        Car("${index + 1}번째 자동차")
+    }
+
+    private val roundHistories = mutableListOf<RoundHistory>()
+
+    fun run(): RaceResult {
+        repeat(raceSetting.totalCountOfRound) { roundNumber: Int ->
+            startRound(cars)
+            saveRoundHistory(roundNumber, cars)
+        }
+
+        return RaceResult(roundHistories)
+    }
+
+    private fun startRound(cars: List<Car>) {
+        Round(cars).start()
```

Comment:
Round 객체를 생성하여 start() 함수를 호출하는 것은 과한 객체화인 것 같아요. 🤔🙂
Round 객체의 start() 메서드 내부 로직을 그대로 가지고와도 충분할 것 같아요. 😉

```suggestion
        cars.forEach { car -> car.move(RandomNumberUtils.getRandomNumber()) }
```

Reply:
넵, 필요없는 객체는 없애는 것이 맞는 것 같습니다.
너무 실생활의 게임에 맞추어 생각하다 보니, 과한 객체화가 생긴 것 같아요 😵‍💫